### PR TITLE
Ea 2499 generate elevation file

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    /**
+     * writes the elevation in a file
+     *
+     * @param directoryLocation folder location
+     */
+    public static void writeElevation(String directoryLocation, String xml) throws InternalServerException {
+    if (!directoryLocation.isEmpty()) {
+        try (PrintWriter writer = new PrintWriter(directoryLocation + WebUserSetFields.SLASH + WebUserSetFields.ELEVATION_FILENAME, StandardCharsets.UTF_8)) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:149]: we should use here the org.apache.commons.io.FileUtils.write(File, string, encoding) and to call file.mkdirs() to ensure that the parent directories are created automatically on the server.

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    /**
+     * get the entity reference from the subject list
+     * @param subject
+     * @return
+     */
+    private String generateQueryText(List<String> subject) {
+      StringBuilder text = new StringBuilder();
+      for(String s : subject) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:122]: actually the entity sets must have only one subject, the for loop is not needed.

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          if (s.contains(WebUserSetFields.AGENT)) {
+              text.append(WebUserSetFields.ELEVATION_AGENT_QUERY);
+          }
+          else if (s.contains(WebUserSetFields.CONCEPT)) {
+              text.append(WebUserSetFields.ELEVATION_CONCEPT_QUERY);
+          }
+          else if(s.contains(WebUserSetFields.TIMESPAN)) {
+              text.append(WebUserSetFields.ELEVATION_TIMESPAN_QUERY);
+          } else {
+              break;
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:132]: this break is not needed when the for loop is removed

**Review by gsergiu — COMMENTED** [review]
I left a few comment that should be addressed

File: set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java [hunk 4]
```diff
@@ -196,15 +194,58 @@ public long countTotalLikes() {
...
+		DBObject fields = new BasicDBObject(WebUserSetModelFields.SUBJECT, 1);
+		fields.put(WebUserSetFields.ITEMS, 1);
+		fields.put(WebUserSetFields.TYPE, 1);
+		// create projection
+		DBObject project = new BasicDBObject(WebUserSetFields.MONGO_PROJECT, fields);
+
+		List<DBObject> pipeline = Arrays.asList(getMongoMatchForPipeLine(WebUserSetFields.TYPE, UserSetTypes.ENTITYBESTITEMSSET.getJsonValue()),
+				                  project);
+
+		Cursor cursor = getDao().getCollection().aggregate(pipeline, aggregationOptions);
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java:226]: changed

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java (hunk 2 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:122]: then we should add the check for the size of subject, it must always be one.

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+          }
+          text.append("\"");
+          text.append(s);
+          text.append("\"");
+
+      }
+      return URLEncoder.encode(text.toString(), StandardCharsets.UTF_8);
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:139]: Now the the output will be something like this ==>  <query text="skos_concept:&quot;http://data.europeana.eu/concept/base/114&quot;">

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java (hunk 1 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:149]: done

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java (hunk 3 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java:132]: yes

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (gsergiu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java]: that is the correct xml output

**Pull request merged**